src/core/server/config/deprecation/core_deprecations.ts

```diff
@@ -38,6 +38,16 @@ const dataPathDeprecation: ConfigDeprecation = (settings, fromPath, log) => {
   return settings;
 };
 
+const xsrfDeprecation: ConfigDeprecation = (settings, fromPath, log) => {
+  if (has(settings, 'server.xsrf.whitelist')) {
+    log(
```

Comment:
NIT: should we also check that `server.xsrf.whitelist` is not empty? (not sure that makes mch sense from a yaml conf file, but...)